session_test.go

```diff
@@ -662,6 +662,31 @@ func (s *S) TestUpdate(c *C) {
 	c.Assert(err, Equals, mgo.ErrNotFound)
 }
 
+func (s *S) TestUpsertMulti(c *C) {
+	session, err := mgo.Dial("localhost:40001")
+	c.Assert(err, IsNil)
+	defer session.Close()
+
+	coll := session.DB("mydb").C("mycoll")
+
+	ns := []int{40, 41, 42, 43, 44, 45, 46}
+	for _, n := range ns {
+		err := coll.Insert(M{"k": 1, "n": n})
+		c.Assert(err, IsNil)
+	}
+
+	// UpsertMulti should _match_ all existing documents, but update one less than that
+	changeInfo, err := coll.UpsertMulti(M{"k": 1}, M{"$set": M{"n": 42}})
+	c.Assert(err, IsNil)
+	c.Assert(changeInfo.Matched, Equals, len(ns))
+	c.Assert(changeInfo.Updated, Equals, len(ns)-1)
```

Comment:
Test looks good, but you probably want to check that the value set is the one you expect in addition to checking the number of documents updated/matched. Have a look at ```TestUpsert``` for an example.